cli/tests/e2e/test_ci.py

```diff
@@ -1735,3 +1741,35 @@ def test_existing_supply_chain_finding(
     post_calls = AppSession.post.call_args_list
     findings_json = post_calls[num_old_post_calls + 1].kwargs["json"]
     assert len(findings_json["findings"]) == 0
+
+
+@pytest.mark.parametrize(
+    "enabled_products",
+    [[], ["product"]],
+    ids=["empty-products", "non-empty-products"],
+)
+def test_enabled_products(
+    enabled_products: List[str],
+    run_semgrep: RunSemgrep,
+    mocker,
+    git_tmp_path_with_commit,
+):
+    """
+    Verify that for any given product, there is a valid output
+    """
+    mocker.patch.object(ScanHandler, "enabled_products", enabled_products)
+
+    result = run_semgrep(
+        options=["ci", "--no-suppress-errors"],
+        target_name=None,
+        strict=False,
+        assert_exit_code=None,
+        env={"SEMGREP_APP_TOKEN": "fake_key"},
```

Comment:
probably need a use_click_runner=True arg here.

Reply:
Fixed here: https://github.com/returntocorp/semgrep/pull/8531

Reply:
Can you give a bit more info on what this ? I see
```
use_click_runner: bool = True,  # TODO: change to False. deprecated! Avoid using! see semgrep_runner.py
```

Reply:
Yes there's a big comment about it in the file defining the function and this argument.
From semgrep_runner.py:
```
# Small wrapper around 'semgrep' useful for writing end-to-end (e2e) tests.
#
# old: This file was originally introduced to optimize the way we were running
# semgrep in tests by using Click and its CliRunner to avoid an extra fork.
# However, with the introduction of osemgrep and the new cli/bin/semgrep (which
# dispatch to osemgrep), we actually want to avoid to use the CliRunner which
# only run pysemgrep code. Otherwise, our e2e tests would not be really end-to
# -end and may not represent what the user would get by using semgrep directly.
# This is why using the CliRunner option is now deprecated.
# TODO: The option is still kept because a few of our tests still rely on
# Click-specific features (e.g., mocking)  that the regular
# call-semgrep-in-a-subprocess does not provide yet.
```